pkg/authorization/reaper/cluster_role.go

```diff
@@ -53,7 +53,7 @@ func (r *ClusterRoleReaper) Stop(namespace, name string, timeout time.Duration,
 		}
 	}
 
-	if err := r.roleClient.ClusterRoles().Delete(name, &metav1.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+	if err := r.roleClient.ClusterRoles().Delete(name, &metav1.DeleteOptions{}); err != nil {
```

Comment:
If the cluster role you're deleting is not found, why should a reaping method (which wants to make sure it was removed) fail?

Reply:
I see. But actually reaping methods for other resources is returning NotFound error. For example, `pod`'s reaper method checks existing before delete it and return NotFound:

https://github.com/kubernetes/kubernetes/blob/430c1a68c8116732511288398ea5cd1b08595e7c/pkg/kubectl/delete.go#L482-L485

I think that user, group and (cluster)role should be same behavior. If I should change this PR in the same manner exactly, I will change this PR as:

~~~
        if _, err := r.roleClient.ClusterRoles().Get(name, metav1.GetOptions{}); err != nil {
                return err
        }
        if err := r.roleClient.ClusterRoles().Delete(name, &metav1.DeleteOptions{}); err != nil {
                return err
        }
~~~~

(But I am thinking above change is a little bit redundant.)

Reply:
> pod's reaper method checks existing before delete it and return NotFound:

That method is wrong.  The power to delete does not imply the power to get.